state/indexer/indexer_service_test.go

```diff
@@ -1,24 +1,115 @@
 package indexer_test
 
 import (
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
 	"testing"
 	"time"
 
+	"github.com/adlio/schema"
+	_ "github.com/lib/pq"
+	dockertest "github.com/ory/dockertest"
+	"github.com/ory/dockertest/docker"
 	"github.com/stretchr/testify/require"
-	db "github.com/tendermint/tm-db"
 
 	abci "github.com/tendermint/tendermint/abci/types"
-	"github.com/tendermint/tendermint/libs/log"
+	tmlog "github.com/tendermint/tendermint/libs/log"
 	indexer "github.com/tendermint/tendermint/state/indexer"
-	blockidxkv "github.com/tendermint/tendermint/state/indexer/block/kv"
-	"github.com/tendermint/tendermint/state/indexer/tx/kv"
+	kvsink "github.com/tendermint/tendermint/state/indexer/sink/kv"
+	psqlsink "github.com/tendermint/tendermint/state/indexer/sink/psql"
 	"github.com/tendermint/tendermint/types"
+	db "github.com/tendermint/tm-db"
+)
+
+var psqldb *sql.DB
+var pSink indexer.EventSink
+
+var (
+	user     = "postgres"
+	password = "secret"
+	port     = "5432"
+	dsn      = "postgres://%s:%s@localhost:%s/%s?sslmode=disable"
+	dbName   = "postgres"
 )
 
+func TestMain(m *testing.M) {
```

Comment:
Please avoid a `TestMain`. Instead use a helper function that could return a struct.

e.g.
```go
func setupDB(t *testing.T, ...) (...) {
  t.Helper()

  // ...
}
```